Tests/SwiftParserTest/AttributeTests.swift

```diff
@@ -965,6 +965,272 @@ final class AttributeTests: ParserTestCase {
     )
   }
 
+  func testABIAttribute() {
+    func abiAttr(_ provider: ABIAttributeArgumentsSyntax.Provider) -> AttributeListSyntax.Element {
+      return .attribute(
+        AttributeSyntax(
+          attributeName: TypeSyntax("abi"),
+          leftParen: .leftParenToken(),
+          arguments: .abiArguments(
+            ABIAttributeArgumentsSyntax(
+              provider: provider
+            )
+          ),
+          rightParen: .rightParenToken()
+        )
+      )
+    }
+
+    assertParse(
+      """
+      @abi(func fn() -> Int)
+      func fn1() -> Int { }
+      """,
+      substructure: FunctionDeclSyntax(
+        attributes: [
+          abiAttr(
+            .function(
+              FunctionDeclSyntax(
+                name: "fn",
+                signature: FunctionSignatureSyntax(
+                  parameterClause: FunctionParameterClauseSyntax {},
+                  returnClause: ReturnClauseSyntax(type: TypeSyntax("Int"))
+                ),
+                body: nil
+              )
+            )
+          )
+        ],
+        name: "fn1",
+        signature: FunctionSignatureSyntax(
+          parameterClause: FunctionParameterClauseSyntax {},
+          returnClause: ReturnClauseSyntax(type: TypeSyntax("Int"))
+        )
+      ) {},
+      experimentalFeatures: [.abiAttribute]
+    )
+
+    assertParse(
+      """
+      @abi(associatedtype AssocTy)
+      associatedtype AssocTy
```

Comment:
Could you split these into multiple `assertParse` calls? That makes debugging a lot simpler if one of the decls fails to parse because you don’t need to reduce the input anymore.

Same below.